lib/Twig/TokenParser/Embed.php

```diff
@@ -22,19 +22,15 @@ public function parse(Twig_Token $token)
 
         list($variables, $only, $ignoreMissing) = $this->parseArguments();
 
-        // inject a fake parent to make the parent() function work
         $stream->injectTokens(array(
             new Twig_Token(Twig_Token::BLOCK_START_TYPE, '', $token->getLine()),
             new Twig_Token(Twig_Token::NAME_TYPE, 'extends', $token->getLine()),
-            new Twig_Token(Twig_Token::STRING_TYPE, '__parent__', $token->getLine()),
+            new Twig_Token(Twig_Token::STRING_TYPE, $parent->getAttribute('value'), $token->getLine()),
```

Comment:
this is broken, because it works only for a ConstantNode, while we support any expression